### PR TITLE
Simplify command objects setup.

### PR DESCRIPTION
**Pull request by calebkiage**

### CODE REVIEW

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 1]
```diff
@@ -10,14 +10,19 @@ namespace Microsoft.Graph.Cli.Core.Http;
 
 public class LoggingHandler : DelegatingHandler
 {
-    public ILogger<LoggingHandler>? Logger { private get; set; }
+    private ILogger<LoggingHandler> _log;
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:13]:
```suggestion
    private readonly ILogger<LoggingHandler> _log;
```

File: src/Microsoft.Graph.Cli.Core/IO/GraphCliClientFactory.cs [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+using System.Net.Http.Headers;
 
 namespace Microsoft.Graph.Cli.Core.IO;
 
 public class GraphCliClientFactory
 {
-    public static IEnumerable<DelegatingHandler> GetDefaultMiddlewaresWithOptions(GraphClientOptions options) => GraphClientFactory.CreateDefaultHandlers(options);
+    /// Microsoft Graph service national cloud endpoints
+    private static readonly Dictionary<string, string> cloudList = new Dictionary<string, string>
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/IO/GraphCliClientFactory.cs:13]:
```suggestion
    private static readonly Dictionary<string, string> cloudList = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
```

File: src/sample/Program.cs [hunk 3]
```diff
@@ -1,96 +1,94 @@
...
 
 namespace Microsoft.Graph.Cli
 {
+    class ServiceBinder<T> : BinderBase<T?> where T: class
+    {
+        protected override T? GetBoundValue(BindingContext ctx)
+        {
+            var host = ctx.GetRequiredService(typeof(IHost)) as IHost;
+
+            return (ctx.GetService(typeof(T)) ?? host?.Services.GetService(typeof(T))) as T;
```
Comment (baywet) [inline src/sample/Program.cs:40]: can you expand on why we need to look for the value at two different places?

File: src/sample/Program.cs [hunk 4]
```diff
@@ -1,96 +1,94 @@
...
+        }
+    }
+
+    public static class Extensions {
+        public static CommandLineBuilder UseRequestAdapter(this CommandLineBuilder builder, Func<InvocationContext, IRequestAdapter> builderFactory) {
+            builder.AddMiddleware(async (context, next) => {
+                var requestAdapter = builderFactory.Invoke(context);
+                if (requestAdapter != null) {
```
Comment (baywet) [inline src/sample/Program.cs:48]:
```suggestion
                if (builderFactory.Invoke(context) is IRequestAdapter requestAdapter) {
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/sample/Program.cs (hunk 3 above)
Comment (calebkiage) [reply src/sample/Program.cs:40]: This is a sample showing how one can use both the .NET Dependency injection library and the System.CommandLine container depending on where they registered the type. Notice that T is a generic type here.

File: src/Microsoft.Graph.Cli.Core/IO/GraphCliClientFactory.cs (hunk 2 above)
Comment (calebkiage) [reply src/Microsoft.Graph.Cli.Core/IO/GraphCliClientFactory.cs:13]: Is this necessary? The keys are constant, and the dictionary is read only....

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 5]
```diff
@@ -79,4 +72,27 @@ private string HeadersToString(in HttpHeaders headers, in HttpHeaders? contentHe
...
+    private async Task<string> ContentToStringAsync(HttpContent? content, CancellationToken cancellationToken = default)
+    {
+        if (content is null) return string.Empty;
+        var responseContent = string.Empty;
+        var isStream = content.Headers.ContentType?.MediaType?.Contains("stream") == true;
+        if (!isStream)
+        {
+            responseContent = await content.ReadAsStringAsync(cancellationToken);
+        }
+        else if (isStream)
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:85]:
```suggestion
        else
```

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 6]
```diff
@@ -79,4 +72,27 @@ private string HeadersToString(in HttpHeaders headers, in HttpHeaders? contentHe
...
+        {
+            string size = string.Empty;
+            if (content.Headers.ContentLength > 0)
+            {
+                size = $"{content.Headers.ContentLength} byte ";
+            }
+
+            responseContent = $"[...<{size}data stream>...]";
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:93]:
```suggestion
            return $"[...<{size}data stream>...]";
```

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 7]
```diff
@@ -79,4 +72,27 @@ private string HeadersToString(in HttpHeaders headers, in HttpHeaders? contentHe
...
+        }
+
+        return responseContent;
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:96]:
```suggestion
```

File: src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs [hunk 8]
```diff
@@ -79,4 +72,27 @@ private string HeadersToString(in HttpHeaders headers, in HttpHeaders? contentHe
...
     }
+
+    private async Task<string> ContentToStringAsync(HttpContent? content, CancellationToken cancellationToken = default)
+    {
+        if (content is null) return string.Empty;
+        var responseContent = string.Empty;
+        var isStream = content.Headers.ContentType?.MediaType?.Contains("stream") == true;
+        if (!isStream)
+        {
+            responseContent = await content.ReadAsStringAsync(cancellationToken);
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs:83]:
```suggestion
            return await content.ReadAsStringAsync(cancellationToken);
```

**Review by baywet — APPROVED** [review]

**Pull request merged**